packages/pnpm-sync-lib/src/pnpmSyncPrepare.ts

```diff
@@ -296,7 +296,11 @@ function processDependencies(
     for (const [dependency, specifier] of Object.entries(dependencies)) {
       if (injectedDependencyToVersion.has(dependency)) {
         const specifierToUse: string = typeof specifier === 'string' ? specifier : specifier.version;
-        injectedDependencyToVersion.get(dependency)?.add(specifierToUse);
+
+        // the injected dependency should always start with file protocol
+        if (specifierToUse.startsWith('file:')) {
+          injectedDependencyToVersion.get(dependency)?.add(specifierToUse);
```

Comment:
@chengcyber we should probably add this to the list of checks for the "monorepo lint" rules